chain-abci/src/app/app_init.rs

```diff
@@ -49,7 +51,7 @@ pub struct ChainNodeState {
 }
 
 /// The global ABCI state
-pub struct ChainNodeApp {
+pub struct ChainNodeApp<T: EnclaveProxy> {
```

Comment:
enclave proxy will be only one, how about giving trait as "new"?

Reply:
could you sketch out the code what you mean by giving a trait as "new"?

Reply:
pub struct ChainNodeApp<T: EnclaveProxy> {
...
pub tx_validator: EnclaveProxy
}
EncalveProxy serves just only for ChainNodeApp

Reply:
also if there can be multiple Proxy like
AmdEnclaveProxy, using generic like <T> seems good.

Reply:
I don't think one can put a trait into a struct like this? This is the error one would get:
```
   --> chain-abci/src/app/app_init.rs:99:5
    |
99 |     pub tx_validator: EnclaveProxy,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `enclave_bridge::EnclaveProxy` cannot be made into an object
    |
    = note: the trait cannot require that `Self : Sized`
```

@devashishdxt any thoughts?

Reply:
it's possible with this style,
not direct member.
pub  tx_validator: &'a EnclaveProxy,

Reply:
it's unclear to me where the EnclaveProxy would be owned / live -- it should live as long as `ChainNodeApp` (which owns it in the current code, so it's trivial).
with the reference, it may be possible to guarantee, but not sure if worth the hassle?

Reply:
no,
it's just a comment.